src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -465,8 +465,36 @@ internal bool TryGetValue(int index, out DateTime value)
             ReadOnlySpan<byte> data = _utf8Json.Span;
             ReadOnlySpan<byte> segment = data.Slice(row.Location, row.SizeOrLength);
 
-            if (JsonHelpers.TryParseAsISO(segment, out DateTime tmp, out int bytesConsumed) &&
-                segment.Length == bytesConsumed)
+            DateTime tmp;
+            int bytesConsumed;
+
+            // Segment needs to be unescaped
+            if (row.HasComplexChildren)
```

Comment:
Generally, this path will be uncommon. Is it possible to move the code in this if-block to a helper method instead?